cmd/src/bin/tikv-ctl.rs

```diff
@@ -2087,8 +2087,8 @@ fn new_security_mgr(matches: &ArgMatches<'_>) -> Arc<SecurityManager> {
         cfg.key_path = key_path.unwrap().to_owned();
     }
 
-    if cipher_file.is_some() {
-        cfg.cipher_file = cipher_file.unwrap().to_owned();
+    if let Some(cipher_file) = cipher_file {
```

Comment:
Interesting. Why can't optimizer catch these things up?

Reply:
The optimizer probably will figure this and similar cases out on its own and eliminate one of the two branches. The lint is more about correctness and readability. Using destructuring is correct by construction where unwrap requires some invariant to hold, and the further away from the unwrap that invariant is the harder it gets to read correctly.